### PR TITLE
add arm template onboarding script

### PR DESCRIPTION
**Pull request by danielohfeld**
Required items, please complete
   
   Change(s):
   - See guidance below

   Reason for Change(s):
   - See guidance below

   Version Updated:
   - Required only for Detections/Analytic Rule templates
   - See guidance below

   Testing Completed:
   - See guidance below

   Checked that the validations are passing and have addressed any issues that are present:
   - See guidance below

# Guidance <- remove section before submitting
-----------------------------------------------------------------------------------------------------------
## Before submitting this PR please ensure that you have read the following sections and filled out the changes, reason for change and testing complete sections:

Thank you for your contribution to the Microsoft Sentinel Github repo.

> Details of the code changes in your submitted PR.  Providing descriptions for pull requests ensures there is context to changes being made and greatly enhances the code review process.  Providing associated Issues that this resolves also easily connects the reason.
   
   Change(s):
   - Add arm template that creates DCE and DCR for a chosen WS.

   Reason for Change(s):
   - Easy customer onboarding for BE pre pp.

   Version updated:
   - No

> The code should have been tested in a Microsoft Sentinel environment that does not have any custom parsers, functions or tables, so that you validate no incorrect syntax and execution functions properly.  If your submission requires a custom parser or function, it must be submitted with the PR.

   Testing Completed:
   - Yes/No/Need Help

_Note: If updating a detection, you must update the version field._

> Before the submission has been made, please look at running the KQL and Yaml Validation Checks locally.
> https://github.com/Azure/Azure-Sentinel#run-kql-validation-locally

   Checked that the validations are passing and have addressed any issues that are present:
   - Yes/No/Need Help
   
   _Note: Let us know if you have tried fixing the validation error and need help._

> References: 
> - [Guidance for Detection checks](https://github.com/Azure/Azure-Sentinel#pull-request-detection-template-structure-validation-check)
> - [General contribution guidance](https://github.com/Azure/Azure-Sentinel/wiki#what-can-you-contribute-and-how-can-you-create-contributions)
> - [PR validation troubleshooting](https://github.com/Azure/Azure-Sentinel#pull-request)

-----------------------------------------------------------------------------------------------------------

### CODE REVIEW

File: DataConnectors/GCP/ArmTemplate/GCPOnboarding.json [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+  "parameters": {
+    "DataCollectionEndpoint_name": {
+      "defaultValue": "DataCollectionEndpoint",
+      "type": "String"
+    },
+    "Location": {
+      "defaultValue": "[resourceGroup().location]",
+      "type": "String"
+    },
+    "DataCollectionRule_name": {
```
Comment (yoziv) [inline DataConnectors/GCP/ArmTemplate/GCPOnboarding.json:13]: why the '_' is that a format you saw somewhere else? did it come for the export  functionally? check out other arm templates to see what the way is it is usually  done

File: DataConnectors/GCP/ArmTemplate/GCPOnboarding.json [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      "location": "[parameters('Location')]",
+      "dependsOn": [
+        "[resourceId('Microsoft.Insights/dataCollectionEndpoints',parameters('DataCollectionEndpoint_name'))]"
+      ],
+      "properties": {
+        "dataCollectionEndpointId": "[resourceId('Microsoft.Insights/dataCollectionEndpoints',parameters('DataCollectionEndpoint_name'))]",
+        "destinations": {
+          "logAnalytics": [
+            {
+              "workspaceResourceId": "[concat('/subscriptions/', subscription().subscriptionId, '/resourceGroups/', resourceGroup().name, '/providers/Microsoft.OperationalInsights/workspaces/', parameters('workspace_name'))]",
```
Comment (yoziv) [inline DataConnectors/GCP/ArmTemplate/GCPOnboarding.json:48]: i prefer to create the workspace resource id as an arm variable and use it here like they say here: https://stackoverflow.com/questions/64803341/concat-variable-with-string-in-azure-arm-template

File: DataConnectors/GCP/ArmTemplate/GCPOnboarding.json [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+              "name": "clv2ws1"
```
Comment (yoziv) [inline DataConnectors/GCP/ArmTemplate/GCPOnboarding.json:49]: move to variable

File: DataConnectors/GCP/ArmTemplate/GCPOnboarding.json [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+            }
+          ]
+        },
+        "dataFlows": [
+          {
+            "streams": [
+              "Microsoft-GCPAuditLogs"
```
Comment (yoziv) [inline DataConnectors/GCP/ArmTemplate/GCPOnboarding.json:56]: move to variable

File: DataConnectors/GCP/ArmTemplate/GCPOnboarding.json (hunk 3 above)
Comment (danielohfeld) [reply DataConnectors/GCP/ArmTemplate/GCPOnboarding.json:49]: why?

Comment (yoziv) [reply DataConnectors/GCP/ArmTemplate/GCPOnboarding.json:49]: for starters it appears twice

**Review by sivanguetta — APPROVED** [review]

**Pull request merged**